### PR TITLE
Integrate Element Templates Runtime Versions

### PR DESCRIPTION
**Pull request by jarekdanielak**
### Proposed Changes

Integrate element templates runtime version compatibility.

Camunda 8 BPMN editor will now pass current runtime version to element templates module. If template is incompatible, warning will be shown in properties panel and problems panel.

To try it out, use the new "Compatibility test" template from `cloud-samples` and try changing template version in XML or Camunda version.

Screenshot 2024-12-12 at 19 53 16

Closes https://github.com/camunda/camunda-modeler/issues/4617
Closes https://github.com/camunda/camunda-modeler/issues/4648
Closes https://github.com/camunda/camunda-modeler/issues/4530
Closes https://github.com/camunda/camunda-modeler/issues/4749

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [x] __Visual demo__ attached
* [x] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: client/src/app/tabs/cloud-bpmn/BpmnEditor.js [hunk 1]
```diff
@@ -849,6 +869,11 @@ export class BpmnEditor extends CachedComponent {
       elementTemplateChooser: false,
       keyboard: {
         bind: false
+      },
+      elementTemplates: {
+        engines: {
+          camundaModeler: version
```
Comment (nikku) [inline client/src/app/tabs/cloud-bpmn/BpmnEditor.js:875]: Let's change this engine to `camundaDesktopModeler`. I assume that the version is the current Desktop Modeler version, i.e. `v5.30.0`.

File: client/src/app/tabs/cloud-bpmn/BpmnEditor.js [hunk 2]
```diff
@@ -245,6 +249,22 @@ export class BpmnEditor extends CachedComponent {
...
+    const elementTemplates = this.getModeler().get('elementTemplates');
+
+    const engines = {
+      ...elementTemplates.getEngines(),
+      camunda: version
+    };
+
+    elementTemplates.setEngines(engines);
+
+    this.loadTemplates();
```
Comment (nikku) [inline client/src/app/tabs/cloud-bpmn/BpmnEditor.js:265]: This should not be necessary, or should it? Templates update automatically when the engine changes.

Comment (barmac) [reply client/src/app/tabs/cloud-bpmn/BpmnEditor.js:265]: I'd expect that it works out of the box without additional calls.

**Comment by jarekdanielak** [comment]
Styles for Incompatible button are missing, but this will be fixed with dependencies update in `camunda-bpmn-js`: https://github.com/camunda/camunda-bpmn-js/pull/396

**Review by nikku — COMMENTED** [review]
Can we / should we test this?

**Comment by jarekdanielak** [comment]
Styles problem is fixed with update to `camunda-bpmn-js@5.3.0`.

> Can we / should we test this?

The test suite for modeling features is quite slim. Do we want to work on extending it as part of this PR?

**Comment by jarekdanielak** [comment]
cc. @lmbateman for final look at the UI.

**Comment by nikku** [comment]
> The test suite for modeling features is quite slim. Do we want to work on extending it as part of this PR?

Let's ensure this does not break accidentally in the future. I don't care which path we take to accomplish this :slightly_smiling_face:

**Comment by nikku** [comment]
Will resolve conflicts now, and test manually.

**Comment by nikku** [comment]
Rebased on `develop`, marked related issues (fixed through the version bumps).

**Review by nikku — APPROVED** [review]
Works on my machine.

@jarekdanielak consider to follow-up (in any way) with [this](https://github.com/camunda/camunda-modeler/pull/4744#issuecomment-2541297755).

**Comment by lmbateman** [comment]
Checked using `npx @bpmn-io/sr bpmn-io/bpmn-js-element-templates#main`

✅ Both buttons look good
✅ Both dropdowns have the right text and options
✅ Lint warnings look good

In general, it looks great to me. Two very minor items to think about:

- "Update" and "Unlink" don't look like actions to me (for all issues, not just Incompatible). [Yana's original designs](https://www.figma.com/design/nrB4K2EXZju7JqJGjODU5L/process-bluepront-dialog?node-id=35-981&t=iEuVFUuykfn90wrB-1) had them in blue. Since this affects things outside this story, I'm not sure it's in scope for now.
- Would it make sense to put a `v` in front of the version? When it's a long string, it's probably fine, but the 1 in the demo confuses me slightly. Version? Error code? Typo? :)

image

**Comment by jarekdanielak** [comment]
@lmbateman the feature is already merged, but I will save your comment as a follow up for later. Thanks for the feedback!

**Pull request merged**